File: Java /Gate-four/level-5/Testingmilkydoughnuts.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;



public class Testingmilkydoughnuts{
        @Test
    public void testForElementsThatAreEven(){
        int[] number = {2,3,4,5,6,7,8,9,10,11};

        int expected = 5;
        int actual = Milkydoughnutstasks.countEvens(number);
        assertEquals(expected, actual);
    }

        @Test
  public void testForTargetingElementsInAnArray(){
        int[] digit = {2,3,4,5,6,7,8,9,10,11};
        int target = 7;
        int expected = 5;
        int actual = Milkydoughnutstasks.linearSearch(digit, target);
        assertEquals(expected, actual);
    }


        @Test
  public void testForPerfectSquaresInAnArray(){
     int number = 5;
     int expected[] = {1, 4, 9, 16, 25};
     int actual [] = Milkydoughnutstasks.squares(number);
     assertArrayEquals(expected, actual);
    }


}

File: Python/Gate-four/level-5/testingMilkyDoughnuts.py
from unittest import TestCase
import milkydoughnuts



class TestForMilkyDoughnutsTasks(TestCase):
    def test_for_even_numbers(self):
        num = [2,3,4,5,6,7,8,9,10,11]
        actual = milkydoughnuts.countEvens(num)
        expected = 5
        self.assertEqual(actual, expected)

    def test_for_linearSearch(self):
        nums = [2,3,4,5,6,7,8,9,10,11]
        target = 7
        actual = milkydoughnuts.linearSearch(nums, target)
        expected = 5
        self.assertEqual(actual, expected)

    def test_for_squares(self):
        number = 5
        actual = milkydoughnuts.squares(number)
        expected = [1, 4, 9, 16, 25]
        self.assertEqual(actual, expected)

